drivers/Aqara/aqara-lock/src/credential_utils.lua

```diff
@@ -0,0 +1,134 @@
+local capabilities = require "st.capabilities"
+local utils = require "st.utils"
+
+local remoteControlStatus = capabilities.remoteControlStatus
+local lockCredentialInfo = capabilities["stse.lockCredentialInfo"]
+
+local credential_utils = {}
+local HOST_COUNT = "__host_count"
+
+credential_utils.save_data = function(driver)
+  driver.datastore:save()
+  driver.datastore:commit()
```

Comment:
this driver saves `credentialInfoTable` like below.
[{"credentialId": 1, "credentialType": "keypad", "userId": "1", "userLabel": "Kim", "userType": "host"}]
This table is used to control user management on plugin.

Reply:
To expand on this Robert, the issue is that there are edge cases where the details can be stored, but not persisted for the 1 hour period, and because that credential info is necessary for communicating with the lock, losing it renders the device non-functional on SmartThings.  We include rate limiting on this API to avoid too excessive use.